src/wallet/wallet.cpp

```diff
@@ -2583,6 +2583,16 @@ void CWallet::UpdatedTransaction(const uint256 &hashTx)
     }
 }
 
+void CWallet::GetScriptForMining(CScript &script)
+{
+    CReserveKey reservekey(this);
```

Comment:
Yes. This current design does not respect the keypool/CReserveKey. Will try a better approach.